src/main/java/seedu/agendum/model/task/ReadOnlyTask.java

```diff
@@ -22,7 +27,12 @@
     default boolean isSameStateAs(ReadOnlyTask other) {
         return other == this // short circuit if same object
                 || (other != null // this is first to avoid NPE below
-                && other.getName().equals(this.getName())); // state checks here onwards
+                && other.getName().equals(this.getName())) // state checks here onwards
+                && (other.isCompleted() == this.isCompleted())
+                && (other.getStartDateTime() == this.getStartDateTime() 
```

Comment:
We will need to compare the completion status, start/end time and deadline for the tests.
But should we allow them to create tasks with same name with different time and completion status?